### DISCUSSION TITLE
Why is a return not allowed here?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#37468] [Help] [answered]
Error I am getting is this:

```
Error:
  x Return statement is not allowed here
```

This should be a standard setup I would think... some thing like this:

import React from "react";

```
const Thanks = () => {
Stuff in here.. then..

/// but this return below is not allowed??? 
  return (
    <div>
      Other stuff in here...
    </div>
  );
};
export default Thanks;
```
complete code of specific page is here:
https://github.com/Elindo586/nextprojectb/blob/main/pages/inventory.js

**Comment by icyJoseph** [comment, marked as answer]
There's some strange syntax going on that file:

```ts
const getInventory = {PartsData}
      .then((response) => {
        if (!response.ok) {
          return response.statusText();
        }
        return response.json();
      })
      .then((invDataArr) => {
        partsList = invDataArr;
      })
      .catch((error) => console.log(error));
  };
```

there's a dangling `}` which breaks the function, and also `{PartsData}` creates an object with a key called PartsData, but why would that result in an object with a then method? I guess PartsData is a promise?

First remove that `}` dangling at line 62, and then fix that `getInventory` function.

**Reply by icyJoseph** [reply]
@Elindo586 did you manage to fix this?
